commands/addResource.ts

```diff
@@ -1,56 +1,270 @@
-    }
-
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    if (!isValidUrl(userInput!)) {
-        await interaction.reply({
-            content: `you are submiting ${userInput} as a new resource. \n Is this link correct? [Y/N]`,
-            ephemeral: true,
-        })
-        const collection = await interaction.channel?.awaitMessages({ filter, max: 1, time: 15000, errors: ['time'] })
-        collection?.map((message: Message) => {
-            switch (message.content.toLowerCase()) {
-                case 'y':
-                    interaction.followUp({
-                        content: '',
-                        ephemeral: true,
-                    })
-                    break;
-                case 'n':
-                    interaction.followUp({
-                        content: 'Try again with the correct resource please!',
-                        ephemeral: true,
-                    })
-                    break;
-                default:
-                    return new Error('something bad happened')
-            }
-        })
+  let contributor;
+  if (!await isContributor(interaction.user)) {
+    await interaction.reply(`it looks like you are not a contributor yet!\nPlease add yourself using: ${inlineCode('/add-contributor')}`)
+    return
+  }
+  else {
+    const contributorResponse = await findContributor(interaction.user);
+    if (contributorResponse) {
+      contributor = contributorResponse.getId();
+    }
+  }
+
+  const url = interaction.options.getString('url');
+  if (!url) {
+    interaction.reply({ content: 'URL required, please try submitting again.', ephemeral: true });
+    return;
+  }
+
+  if (!isValidUrl(url)) {
+    interaction.reply({ content: 'Invalid URL provided, please check it before submitting again.', ephemeral: true });
+    return;
+  }
+
+  const foundResouce = await findResourceByUrl(url);
+  if (foundResouce) {
+    interaction.reply({ content: 'A resource with this URL already exists.', ephemeral: true });
+    return;
+  }
+
+  await interaction.deferReply();
+
+  const resource = getSanitizedResourceInfo(interaction);
+  resource.contributor = contributor;
+
+  const resourceEmbed = buildEmbed(resource);
+  const tags = await readTags();
+  const tagsOptions = tags.map(tag => ({ label: tag.name, value: tag.id }));
+  const tagsRow = new MessageActionRow().addComponents(
+    new MessageSelectMenu()
+      .setCustomId('tags')
+      .setPlaceholder('Select tags')
+      .setMaxValues(Math.min(tagsOptions.length, 25))
+      .addOptions(tagsOptions),
+  );
+
+  const blockchain = await readBlockchain();
+  const bcOptions = blockchain.map(bc => ({ label: bc.name, value: bc.id }));
+  const bcRow = new MessageActionRow().addComponents(
+    new MessageSelectMenu()
+      .setCustomId('blockchain')
+      .setPlaceholder('Select blockchain')
+      .setMaxValues(Math.min(bcOptions.length, 25))
+      .addOptions(bcOptions),
+  );
```

Comment:
I'd like to propose renaming it to blockchainOptions to be consistent with the other fields such as `categoryOptions`, although that's certainly non-blocking.

I'd also like to suggest adding an empty field to all of the optional options, as you *have* to answer all of these drop-downs to proceed, but there may be cases that none of the tags are appropriate, and it feels like a bad UX to have to abort all of the progress you made filling out the fields, to have to go back and create a new category/tag/whatever and re-do the process.

This is just my first reaction, happy to discuss further.

Reply:
Good point. Let me check that, I did have it initially that you didn't have to answer all of them for the option to finish to appear. Then I added in more select menus.